### DISCUSSION TITLE
Document `pnpm prune`

### DISCUSSION THREAD

**Discussion by Kinrany** [pnpm/pnpm#3527] [Q&A] [answered]
What does `pnpm prune` actually do?

I assume it's similar to `npm prune`, but `pnpm` is sufficiently different that specific behavior is not obvious. Plus understanding its function requires reading `npm` docs.

Right now `pnpm` docs just [say](https://pnpm.io/cli/prune) that it "removes unnecessary packages". What exactly does "unnecessary" mean?

I started looking at it because I want to delete all cached files for a project without deleting the project itself -- both `node_modules` and all the packages in store not used by other projects, similar to Rust's `cargo clean`. I assume `pnpm prune` doesn't do that?

**Comment by zhangzhuang15** [comment, marked as answer]
Follow this link: https://github.com/orgs/pnpm/discussions/4546, it will help you.

**Reply by Kinrany** [reply]
Thank you!

> [It] reads every file in the content-addressable store and removes those that have only 1 reference. 1 reference means that the file is not linked from any project on the computer.
>
> https://github.com/pnpm/pnpm/blob/main/packages/package-store/src/storeController/prune.ts
